src/app/pages/products/products-list/product-card/product-card.component.html

```diff
@@ -43,10 +43,16 @@ <h2 class="description__title">{{product.title | titlecase}}</h2>
                 <a routerLink="/products/{{product.id}}">Product Details</a>
                 <img src="assets/images/svg/ic-arrow-right.svg" alt="arrow">
             </button>
-            <button class="card__button wish-button">
-                <img src="assets/images/svg/ic-heart.svg" alt="heart">
+            <button class="card__button wish-button" *ngIf="!existInWishlist" (click)="addToWishList(product)">
```

Comment:
It is a bad idea - used getter or method inside template without 'onPush' changeDetectStrategy, this way have a worse performer in your app. Please, create a public field or use async pipe or change your detect strategy to onPush instead of the default.